src/lib/ui/thermostat/TemperatureControl.tsx

```diff
@@ -0,0 +1,126 @@
+import { type ChangeEventHandler, useEffect, useRef } from 'react'
+
+import { TemperatureAddIcon } from 'lib/icons/TemperatureAdd.js'
+import { TemperatureSubtractIcon } from 'lib/icons/TemperatureSubtract.js'
+
+const DEFAULT_MIN = 50
+const DEFAULT_MAX = 90
+
+interface TemperatureControlProps {
+  variant: 'heat' | 'cool'
+  temperature: number
+  onChange: (temperature: number) => void
+  min?: number
+  max?: number
+  temperatureUnit?: 'fahrenheit' | 'celsius'
+}
+
+export function TemperatureControl({
+  variant,
+  temperature,
+  onChange,
+  min = DEFAULT_MIN,
+  max = DEFAULT_MAX,
+  temperatureUnit = 'fahrenheit',
+}: TemperatureControlProps): JSX.Element {
+  const increment = (): void => {
+    onChange(Math.min(temperature + 1, max))
+  }
+
+  const decrement = (): void => {
+    onChange(Math.max(temperature - 1, min))
+  }
+
+  return (
+    <div className='seam-temperature-control'>
+      <button className='seam-temperature-stepper-button' onClick={decrement}>
+        <TemperatureSubtractIcon />
+      </button>
+      <RangeSlider
+        variant={variant}
+        min={min}
+        max={max}
+        temperature={temperature}
+        onChange={(temperature) => {
+          onChange(temperature)
+        }}
+        temperatureUnit={temperatureUnit}
+      />
+      <button className='seam-temperature-stepper-button' onClick={increment}>
+        <TemperatureAddIcon />
+      </button>
+    </div>
+  )
+}
+
+interface RangeSliderProps {
+  variant: 'heat' | 'cool'
+  temperature: number
+  onChange: (temperature: number) => void
+  min: number
+  max: number
+  temperatureUnit: 'fahrenheit' | 'celsius'
+}
+
+function RangeSlider({
+  variant,
+  temperature,
+  onChange,
+  min,
+  max,
+  temperatureUnit,
+}: RangeSliderProps): JSX.Element {
+  const wrapRef = useRef<HTMLDivElement>(null)
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    if (wrapRef.current != null && inputRef.current != null) {
+      const wrap = wrapRef.current
+      const input = inputRef.current
+
+      wrap.style.setProperty('--temperature-current', input.value)
+      wrap.style.setProperty(
+        '--temperature-min',
+        input.min === '' ? String(DEFAULT_MIN) : input.min
```

Comment:
When would `input.min` or `input.max` be empty string?

Reply:
Ahh, nice catch—old logic